extractor/src/main/java/org/schabi/newpipe/extractor/utils/ManifestCreatorCache.java

```diff
@@ -0,0 +1,301 @@
+     */
+    public void resetMaximumSize() {
+        this.maximumSize = DEFAULT_MAXIMUM_SIZE;
+    }
+
+    /**
+     * Gets the current clear factor of the cache, used when the cache limit size is reached.
+     *
+     * @return the current clear factor of the cache
+     */
+    public double getClearFactor() {
+        return clearFactor;
+    }
+
+    /**
+     * Sets the clear factor of the cache, used when the cache limit size is reached.
+     *
+     * <p>
+     * The clear factor must be a double between {@code 0} excluded and {@code 1} excluded.
+     * </p>
+     *
+     * <p>
+     * Note that it will be only used the next time the cache size limit is reached.
+     * </p>
+     *
+     * @param clearFactor the new clear factor of the cache
+     * @throws IllegalArgumentException if the clear factor passed a parameter is invalid
+     */
+    public void setClearFactor(final double clearFactor) {
+        if (clearFactor <= 0 || clearFactor >= 1) {
+            throw new IllegalArgumentException("Invalid clear factor");
+        }
+
+        this.clearFactor = clearFactor;
+    }
+
+    /**
+     * Resets the clear factor to its {@link #DEFAULT_CLEAR_FACTOR default value}.
+     */
+    public void resetClearFactor() {
+        this.clearFactor = DEFAULT_CLEAR_FACTOR;
+    }
+
+    /**
+     * Reveals whether an object is equal to a {@code ManifestCreator} cache existing object.
+     *
+     * @param obj the object to compare with the current {@code ManifestCreatorCache} object
+     * @return whether the object compared is equal to the current {@code ManifestCreatorCache}
+     * object
+     */
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        final ManifestCreatorCache<?, ?> manifestCreatorCache =
+                (ManifestCreatorCache<?, ?>) obj;
+        return maximumSize == manifestCreatorCache.maximumSize
+                && Double.compare(manifestCreatorCache.clearFactor, clearFactor) == 0
+                && concurrentHashMap.equals(manifestCreatorCache.concurrentHashMap);
+    }
+
+    /**
+     * Returns a hash code of the current {@code ManifestCreatorCache}, using its
+     * {@link #maximumSize maximum size}, {@link #clearFactor clear factor} and
+     * {@link #concurrentHashMap internal concurrent hash map} used as a cache.
+     *
+     * @return a hash code of the current {@code ManifestCreatorCache}
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(maximumSize, clearFactor, concurrentHashMap);
+    }
+
+    /**
+     * Returns a string version of the {@link ConcurrentHashMap} used internally as the cache.
+     *
+     * @return the string version of the {@link ConcurrentHashMap} used internally as the cache
```

Comment:
Btw, comments like this one can be made on one line. It doesn't make sense to repeat twice the exact same thing: the `@return` attribute is made exactly for that.
```suggestion
     * @return the string version of the {@link ConcurrentHashMap} used internally as the cache
```